core/src/components/button/button.scss

```diff
@@ -210,11 +211,20 @@
 }
 
 
+// Button Slots
+// --------------------------------------------------
+
+::slotted([slot=start]),
+::slotted([slot=end]) {
+  flex-shrink: 0;
+}
+
+
 // Button Icons
 // --------------------------------------------------
 
 ::slotted(ion-icon) {
-  font-size: 1.4em;
+  font-size: 1.35em;
```

Comment:
`font-size: 1.4em` with `min-height: 25px` was triggering a browser rendering issue where the text in a button was misaligned with the icon. We did not notice this before because the button height was `height: 25px`. Now that the height of the button is only the _minimum_ height we are now seeing this bug. The issue appears to be that the height of the icon is causing the text to be misaligned.

We tried to fix it by setting `min-height: 25.19px` on `ion-button`. This fixed the issue for Chrome and Firefox, but it made the problem worse for Safari. Instead, we found that by using a slightly smaller icon size we can get the icon and the text to be aligned.

Reply:
It also turns out that the icons were not aligned with the text in `main` even with `height: 25px`. Looks like `min-height: 25px` made the issue worse enough that we are noticing it now.

| main | branch |
| - | - |
| ![image](https://github.com/ionic-team/ionic-framework/assets/2721089/3706380b-1ad4-4273-a58d-1c669495051b) | ![image](https://github.com/ionic-team/ionic-framework/assets/2721089/d8c641ee-b027-4476-9291-5323961a7f47) |

In `main`, there are 2px of blank space at the bottom of the text relative to the bottom of the icon. In `branch`, there is 1px of blank space above and below the text.

Reply:
Note that some buttons may have an "off by 1" alignment due to fonts that are odd (i.e. `13px`). This is a problem in `main` too, so this change does not worsen the issue.

Reply:
The small items in an item on MD also have a font size of 12px which results in 1.4 * 12 = 16.8 (rounded to 17px). I considered changing this to be 13px to avoid changing the icon size. However, this caused larger visual regressions than I was comfortable with, so I opted to stick with the above approach.